### ISSUE TITLE
luci-mod-network: Share WiFi access credential with QRcode

### ISSUE THREAD

**Issue by pgaufillet** [openwrt/luci#2433]
Hi LuCI team!

It would be great if the Wifi module of LuCI could display a QRcode for sharing Wifi credentials.
See [Qifi](https://github.com/evgeni/qifi) for example.

Thanks for your great job!

**Comment by leonghui** [comment]
Interesting suggestion. Instead of Javascript, [qrencode ](https://github.com/openwrt/packages/tree/master/libs/qrencode) is currently used by 3 LuCI packages to generate QR codes:
- luci-app-wireguard
- luci-app-fwknopd
- luci-app-travelmate

**Comment by greyltc** [comment]
This would be great. Android Q and above has joining wifi via qr code baked into the os in a prominent way.

**Comment by feckert** [comment]
What format/config file should the QR-Code have?

**WIFI:S:\<ssid\>;T:\<encryption\>;P:\<password\>;H:\<hide\>**

**Comment by plm** [comment]
Travelmate has an example of [the `qrencode` arguments](https://github.com/openwrt/luci/blob/29e4b397da2126ef7a433ffb8dde47d879a47ad6/applications/luci-app-travelmate/luasrc/view/travelmate/runtime.htm#L200) required to generate a QR code for the Wifi credentials.

**Comment by hmh** [comment]
> This would be great. Android Q and above has joining wifi via qr code baked into the os in a prominent way.

Also iOS 11+, in the built-in camera application.

**Comment by OW87** [comment]
I would also like to see this this feature for the Wifi networks.

If we could add a QR Code on the line for every wifi like this :
![image](https://user-images.githubusercontent.com/15172121/153765384-ec8da35f-9c16-4e32-8d31-c493d3ee274f.png)

@feckert : Are you working on this feature ? 
Thank you for your reply.

**Comment by jonathan-annett** [comment]
If it helps the luci developers, here is a simple page i wrote for creating QR codes (it uses a few libs - it's pure javascript, happens client side in browser )

**you might need to host these on the router so i'm not sure if size constraints prevent using this approach.** 

I don't suggest you use it like this for security reasons but a working sample QR can be generated by appending a string to the url ```https://qr.1mb.site/?code=``` :  [sample link](https://qr.1mb.site/?code=WIFI:T:WPA;S:myssid;P:mysecret;;)
 
To implement it in the luci ui, have a look at this source https://qr.1mb.site/index.js (it parses the passed in url as a string and displays it verbatim. wifi QR codes have a specific syntax, which you can see after "code=" in the url - which for simplity in this demo isn't urlencoded. as you'll be generating it locally in the browser url encoding is irrelevant, i mention it here as obviously complex passwords can't be demoed via this generic site without url ecoding)

the specific syntax for the qr to pass into the library function is documented [here](https://pocketables.com/2022/01/how-to-format-that-wifi-qr-code-in-plain-text.html) however for a wpa2 protected ssid it is basically ```WIFI:S:ssid;T:WPA;P:password;;```

 
in a nutshell, i'm pulling in a few files via a cdn:  

```https://cdnjs.cloudflare.com/ajax/libs/qrcodejs/1.0.0/qrcode.min.js```  - [link](https://cdnjs.cloudflare.com/ajax/libs/qrcodejs/1.0.0/qrcode.min.js) - [info](https://cdnjs.com/libraries/qrcodejs) - [github](https://github.com/davidshimjs/qrcodejs?utm_source=cdnjs&utm_medium=cdnjs_link&utm_campaign=cdnjs_library)


```https://cdnjs.cloudflare.com/ajax/libs/jquery/1.8.3/jquery.min.js``` - [link](https://cdnjs.cloudflare.com/ajax/libs/jquery/1.8.3/jquery.min.js) - library used by qrcode - [info](https://cdnjs.com/libraries/jquery/1.8.3)

and calling them from index.js

**Comment by hmh** [comment]
fwknock support already generates QRcodes from within Luci, whatever is done should be shared between the two packages for space savings.

**Comment by jeffsf** [comment]
Would be a great "standard" feature as it is friendly to non-technical users and encourages the use of more secure passwords than "Oh, what is easy to type and tell people?"

**Comment by systemcrash** [comment]
Closed by 805b2db67012ee76ea8baf57cf83c4dc31b72832
